### DISCUSSION TITLE
What are the (current) requirements for producing a `bootc` compatible base container image?

### DISCUSSION THREAD

**Discussion by miabbott** [bootc-dev/bootc#207] [Q&A]
This came up in discussions at an internal F2F...what is required to produce a `boot` compatible base container image?

The conversations I had seemed to indicate a requirement on the content of the container image (i.e. including `kernel` and `bootc`), but also how the container image was produced.

- Do we require users to produce base container images via `rpm-ostree compose image`?  
- Can users create a base container image by using `ostree container encapsulate` on an existing ostree commit?  
- Does the ostree commit need to be generated using "unified core" mode?

cc: @runcom @travier

**Comment by travier** [comment]
As we want bootc images to have support for [bootupd](https://github.com/coreos/bootupd), we need the images to be built using unified core mode.

If we want to remove that dependency (but I don't think we should), then this is https://github.com/coreos/bootupd/issues/441.

Thus while technically an `rpm-ostree compose tree --unified-core=true + ostree container encapsulate` should work, I think we should standardize on `rpm-ostree compose image` being the only supported option to build bootc base images.

**Reply by vrothberg** [reply]
I like that idea.

Once the conversation has settled, please make sure that the (current) recommended way of building base images is documented.

**Reply by runcom** [reply]
I think this is here https://containers.github.io/bootc/bootc-images/#building-bootc-compatible-base-images (except the standardization and a warning that osbuild is *not* producing unified-core/bootupd compatible images)

**Reply by miabbott** [reply]
Heh, I wrote those docs  :smile: 

I started this discussion to confirm those assumptions and get better clarity.

We have places in the docs where we allude to non-ostree backends in the future...should that be a factor in how we define what a bootc base image is? Or just address the reality of today?

**Reply by travier** [reply]
From my perspective, while we may plan for a future where we don't have ostree at all anymore, right now, it's still there, so we should mention it as it helps understand what's going on.

**Comment by cgwalters** [comment]
https://github.com/containers/bootc/pull/215 will change this to "almost none".
